src/gui/symbology-ng/qgssymbollayerv2widget.cpp

```diff
@@ -45,6 +45,14 @@
 #include <QStandardItemModel>
 #include <QSvgRenderer>
 #include <QMessageBox>
+#include <QTextDocument>
+
+#define QGS_BLOCK_SIGNAL_DURING_CALL( widget, call )\
```

Comment:
@vmora as far as I can tell, the QGS_BLOCK_SIGNAL_DURING_CALL changes are unrelated to the data defined button changes. Can you please split this change off into its own PR, so that the merits or otherwise of this change can be discussed separately.